src/lib/annotations/Annotator.js

```diff
@@ -241,93 +240,112 @@ class Annotator extends EventEmitter {
-                buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_CANCEL).classList.add(CLASS_HIDDEN);
-                postButtonEl.classList.add(CLASS_HIDDEN);
-            }
+            const postButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_POST);
+            annotatorUtil.hideElement(postButtonEl);
+        }
 
-            this.unbindModeListeners(); // Disable draw mode
-            this.bindDOMListeners(); // Re-enable other annotations
+        this.unbindModeListeners(mode); // Disable mode
+        this.bindDOMListeners(); // Re-enable other annotations
+    }
 
-            // Otherwise enter draw mode
-        } else {
-            this.emit('annotationmodeenter', TYPES.draw);
-            this.annotatedElement.classList.add(CLASS_ANNOTATION_DRAW_MODE);
+    /**
+     * Enables the specified annotation mode
+     *
+     * @param {string} mode - Current annotation mode
+     * @param {HTMLElement} buttonEl - Annotation button element
+     * @return {void}
+     */
+    enableAnnotationMode(mode, buttonEl) {
+        this.emit('annotationmodeenter', mode);
+        this.annotatedElement.classList.add(CLASS_ANNOTATION_MODE);
+        if (buttonEl) {
+            buttonEl.classList.add(CLASS_ACTIVE);
 
-            if (buttonEl) {
+            const drawEnterEl = buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_ENTER);
+            const drawCancelEl = buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_CANCEL);
+            const postButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_POST);
+            if (postButtonEl && drawEnterEl && drawCancelEl) {
                 buttonEl.classList.add(CLASS_ACTIVE);
-                buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_ENTER).classList.add(CLASS_HIDDEN);
-                buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_CANCEL).classList.remove(CLASS_HIDDEN);
+                drawEnterEl.classList.add(CLASS_HIDDEN);
+                drawCancelEl.classList.remove(CLASS_HIDDEN);
                 postButtonEl.classList.remove(CLASS_HIDDEN);
             }
-
-            const thread = this.createAnnotationThread([], {}, TYPES.draw);
-            this.unbindDOMListeners();
-            this.bindDrawModeListeners(thread, postButtonEl);
         }
+
+        this.unbindDOMListeners(); // Disable other annotations
+        this.bindModeListeners(mode); // Enable mode
+    }
+
+    /**
+     * Exits all annotation modes except the specified mode
+     *
+     * @param {string} mode - Current annotation mode
+     * @return {void}
+     */
+    exitAnnotationModes(mode) {
+        Object.keys(this.modeButtons).forEach((type) => {
+            if (mode === type) {
+                return;
+            }
+
+            const buttonSelector = this.modeButtons[type].selector;
+            const buttonEl = event.target || this.previewUI.getAnnotateButton(buttonSelector);
```

Comment:
I don't think event is in the scope of this function. Might be a good idea to have modeButtons hold a tuple of modeName -> (selector, element). Otherwise previewUI.getAnnotateButton should be good enough.